### PR TITLE
Basic-UI

### PR DESCRIPTION
**Pull request by ikeia**
Plugins:
- Added tween library

Player Stats:
- Just a bunch of setter and getters with references that update UI and relevant values when the values are changed
- Update function has some testing inputs for testing values (Health, Speed, Reloading)

Health Bar:
- updates once every time Health in Player Stats is changed
- Instant health change with gradual tween layered underneath
- Bar shake on health lost

Ammo Bar:
- Dynamically changes based on ammo capacity
- Reloads on empty
- Put in fire rate kind of

Place holder reticle put in

### CODE REVIEW

**Comment by MaXeraph** [comment]
Are some of the plugins able to be removed ? I'm actually unable to read the unity files cause there's a lot of plugin data

**Comment by ikeia** [comment]
I can get rid of the UI one but I used the tweening to ease a lot of values

**Comment by MaXeraph** [comment]
Mhm maybe could you give us the list (and links) of files that you modified ? we can just look in there individually then

**Comment by MaXeraph** [comment]
Nice lmfao one plugin, should be good to review now!

**Comment by ikeia** [comment]
relevant scripts:

[Health](https://github.com/MaXeraph/Adrenarun/blob/30464904c2c78c06790b6209747d1b567fa95473/Assets/Scripts/UI/UI_health.cs)

[Ammo](https://github.com/MaXeraph/Adrenarun/blob/30464904c2c78c06790b6209747d1b567fa95473/Assets/Scripts/UI/AmmoUI.cs)

[Player Stats](https://github.com/MaXeraph/Adrenarun/blob/30464904c2c78c06790b6209747d1b567fa95473/Assets/Scripts/Player/PlayerStats.cs)

[Movement](https://github.com/MaXeraph/Adrenarun/blob/30464904c2c78c06790b6209747d1b567fa95473/Assets/Scripts/Player/Movement.cs)

[Player Controls](https://github.com/MaXeraph/Adrenarun/blob/30464904c2c78c06790b6209747d1b567fa95473/Assets/Scripts/Player/PlayerControls.cs)

File: Assets/Scripts/Player/PlayerControls.cs [hunk 1]
```diff
@@ -48,7 +51,12 @@ void Update()
 
         if (Input.GetMouseButton(0))
         {
-            ShootControl.Shoot(_gun);
+            if (Time.time >= last_shot + stats.fireRate)
+            {
+                last_shot = Time.time;
+                ShootControl.Shoot(_gun);
+                stats.Ammo -= 1;
+            }
```
Comment (MaXeraph) [inline Assets/Scripts/Player/PlayerControls.cs:59]: I'm already working on this part so you can just revert back to 
```suggestion
                ShootControl.Shoot(_gun);
``` 
The firerate is being incorporated in a Weapon class

File: Assets/Scripts/Player/PlayerControls.cs [hunk 2]
```diff
@@ -8,8 +8,10 @@ public class PlayerControls : MonoBehaviour
     private GameObject _player;
     private GameObject _camera;
     private GameObject _gun;
+    private PlayerStats stats;
 
     private Vector3 _velocity;
+    private float last_shot;
```
Comment (MaXeraph) [inline Assets/Scripts/Player/PlayerControls.cs:14]: Can remove this as well

File: Assets/Scripts/Player/PlayerStats.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using DG.Tweening;
+
+public class PlayerStats : MonoBehaviour
```
Comment (MaXeraph) [inline Assets/Scripts/Player/PlayerStats.cs:6]: For now I'd say keep this and we can pair it out to incorporate other things

**Review by MaXeraph — CHANGES_REQUESTED** [review]
Left some comments, other wise i think its good

**Comment by MaXeraph** [comment]
i can resolve these conflicts and merge in later

File: Assets/Scripts/Player/bob.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+using UnityEngine;
+
+public class bob : MonoBehaviour
+{
+    Vector3 startPos;
+    public float amplitude = 10f;
+    public float period = 1f;
+    private Transform body;
+
+    protected void Start()
```
Comment (zhuandy42) [inline Assets/Scripts/Player/bob.cs:12]: FPS games should not have a bob as the character moves, it throws off the aim of the player

File: Assets/Scripts/Player/PlayerStats.cs (hunk 3 above)
Comment (zhuandy42) [reply Assets/Scripts/Player/PlayerStats.cs:6]: we're trying to stick to a code structure we've outlined in our UML diagram linked in the README. we'll probably need a big refactor to incorporate the behavior in this script according to our structure.

**Comment by ikeia** [comment]
UI Scripts:
[UI Manager](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/UI/UIManager.cs)
[Ammo UI](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/UI/AmmoUI.cs)
[Health UI](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/UI/UI_health.cs)
[Compass UI](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/UI/CompassUI.cs)

Modified Scripts:
[Player Controls](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/Player/PlayerControls.cs)
[Stats](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/Stats.cs)
[Attack Behavior](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/Weapons/AbstractAttackBehaviour.cs)
[Weapon](https://github.com/MaXeraph/Adrenarun/blob/063a401c54291cab74a632fa4d46c8e1ce60efeb/Assets/Scripts/Weapons/Weapon.cs)

File: Assets/Scripts/Player/PlayerControls.cs [hunk 5]
```diff
@@ -4,33 +4,35 @@
...
 
     void Start()
     {
         Cursor.lockState = CursorLockMode.Locked;
         _player = GameObject.FindWithTag("Player");
+
         _camera = Camera.main;
-        
-        
+        //setWeapon(new BulletAttackBehaviour(EntityType.PLAYER), 0.2f, 16, 1f);
```
Comment (zhuandy42) [inline Assets/Scripts/Player/PlayerControls.cs:21]: please avoid pushing non-labelled commented code to master

File: Assets/Scripts/Stats.cs [hunk 6]
```diff
@@ -17,24 +17,49 @@ public Stats(float maxHealth = 100f)
...
+        if (currentHealth <= 0)
+        {
+            Destroy(gameObject);
+        }
+    }
+
     public void takeDamage(float damage)
     {
-        currentHealth -= damage;
+        setHealth(currentHealth - damage);
```
Comment (zhuandy42) [inline Assets/Scripts/Stats.cs:62]: we can revisit the structure of heal and takeDamage, they're pretty much the same method

**Review by zhuandy42 — APPROVED** [review]
some minor things, but just remember to either label why commented code is left in the file or avoid leaving it in altogether. the conflicts are just a small issue as well. lgtm!

**Pull request merged**